mmu-to-hats/catalog_functions/base_transformer.py

```diff
@@ -0,0 +1,34 @@
+import h5py
+import pyarrow as pa
+import numpy as np
+from pathlib import Path
+from typing import Union
+from catalog_functions.utils import np_to_pyarrow_array
+from abc import ABC, abstractmethod
+
+try:
+    from upath import UPath
+except ImportError:
+    UPath = None
+
+
+class BaseTransformer(ABC):
+    """Transforms catalog HDF5 files to PyArrow tables with proper schema."""
+
+    @classmethod
```

Comment:
Should we simplify it and just make them all to be methods? It can help with easier reusability over different datasets